sdk/database/helper/connutil/cloudsql.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package connutil
+
+import (
+	"fmt"
+
+	"cloud.google.com/go/cloudsqlconn"
+	"cloud.google.com/go/cloudsqlconn/postgres/pgxv4"
+)
+
+func (c *SQLConnectionProducer) getCloudSQLDriverName() (string, error) {
+	var driverName string
+	// using switch case for future extensibility
+	switch c.Type {
+	case dbTypePostgres:
+		driverName = cloudSQLPostgres
+	default:
+		return "", fmt.Errorf("unsupported DB type for cloud IAM: %s", c.Type)
+	}
+
+	return driverName, nil
+}
+
+func (c *SQLConnectionProducer) registerDrivers(driverName string, credentials string) (func() error, error) {
```

Comment:
nit: Do we need to pass `driverName` and `credentials` to this method? They should be available on the method receiver.

Reply:
Sorry for the delayed response! We opted to keep the function as is for future extensibility, with the idea that different types of credentials can be passed in here in the future. Currently since the connection producer only exposes the `ServiceAccountJSON` field, using the connection producer parameter would tie this function to only GCP use-cases. It is definitely possible to go either way with some additional pieces, but keeping this in for now and can update in the future depending on future design choices :)